dipy/reconst/cti.py

```diff
@@ -0,0 +1,634 @@
+        A GradientTable class instance for first DDE diffusion epoch
+        gtab2: dipy.core.gradients.GradientTable
+        A GradientTable class instance for second DDE diffusion epoch
+
+        fit_method : str or callable
+
+
+        args, kwargs : arguments and key-word arguments passed to the
+        fit_method.
+
+        """
+        self.gtab1 = gtab1
+        self.gtab2 = gtab2
+        self.args = args
+        self.kwargs = kwargs
+        self.common_fit_method = not callable(fit_method)
+
+        if self.common_fit_method:
+            try:
+                self.fit_method = common_fit_methods[fit_method]
+            except KeyError:
+                msg = '"' + str(fit_method) + '" is not a known fit method. The'
+                msg += ' fit method should either be a function or one of the'
+                msg += ' common fit methods.'
+                raise ValueError(msg)
+
+        self.args = args
+        self.kwargs = kwargs
+
+        self.min_signal = self.kwargs.pop('min_signal', None)
+        if self.min_signal is None:
+            self.min_signal = MIN_POSITIVE_SIGNAL
+        elif self.min_signal <= 0:
+            msg = "The `min_signal` key-word argument needs to be strictly"
+            msg += " positive."
+            raise ValueError(msg)
+
+        self.design_matrix = design_matrix(self.gtab1, self.gtab2)
+        self.inverse_design_matrix = np.linalg.pinv(self.design_matrix)
+
+        tol = 1e-6
+        self.min_diffusivity = tol / -self.design_matrix.min()
+        # self.convexity_constraint = fit_method in {'CLS', 'CWLS'}
+        # if self.convexity_constraint:
+        #     self.cvxpy_solver = self.kwargs.pop('cvxpy_solver', None)
+        #     self.convexity_level = self.kwargs.pop('convexity_level', 'full')
+        #     msg = "convexity_level must be a positive, even number, or 'full'."
+        #     if isinstance(self.convexity_level, str):
+        #         if self.convexity_level == 'full':
+        #             self.sdp_constraints = load_sdp_constraints('dki')
+        #         else:
+        #             raise ValueError(msg)
+        #     elif self.convexity_level < 0 or self.convexity_level % 2:
+        #         raise ValueError(msg)
+        #     else:
+        #         if self.convexity_level > 4:
+        #             msg = "Maximum convexity_level supported is 4."
+        #             warnings.warn(msg)
+        #             self.convexity_level = 4
+        #         self.sdp_constraints = load_sdp_constraints(
+        #             'dki', self.convexity_level)
+        #     self.sdp = PositiveDefiniteLeastSquares(22, A=self.sdp_constraints)
+
+        self.weights = fit_method in {'WLS', 'WLLS', 'UWLLS', 'CWLS'}
+        self.is_multi_method = fit_method in ['WLS', 'OLS', 'UWLLS', 'ULLS',
+                                              'WLLS', 'OLLS', 'CLS', 'CWLS']
+
+    def fit(self, data, mask=None): #here data is cti_params of shape : (n, 48) 
```

Comment:
We want to fit multi voxels so add the line @multi_voxel_fit in the top of this function.